### DISCUSSION TITLE
Hacktoberfest is here!

### DISCUSSION THREAD

**Discussion by RafaelGSS** [fastify/fastify#2583] [General] [answered]
Hello folks!

The last year [we joined at Hacktoberfest 2019](https://github.com/fastify/fastify/issues/1875) adding the `hacktoberfest` flag to some issues labeled as `good first issue`.

Here I would like to propose the same approach, but I feel that is good to understand more about the first contributors what they feel that we are missing on `fastify` to be more "easy to help".

Go on?

**Comment by Eomm** [comment]
I like this initiative and it let users find fastify searching for that tag.

> but I feel that is good to understand more about the first contributors what they feel that we are missing on fastify to be more "easy to help".

I'm not sure I get the point here. Right now the answer is "better docs, more tutorials" from my point of view

**Reply by RafaelGSS** [reply]
Yes, my suggestion is to understand _"what part of our doc is not clear enough"_ or _"what good tutorials for new contributors are missing?"_

**Comment by Ethan-Arrowood** [comment]
What if some of our hacktoberfest issues were to create tutorials?

**Reply by mcollina** [reply]
That would be great as well. Every PRs count.

**Reply by Ethan-Arrowood** [reply]
As time is valuable I've never found a great opportunity to build my own tutorial for fastify, but I'd be heavily inclined to help other's with their tutorials! Proof-reading, testing it out, providing input. Maybe we as maintainers could brainstorm what we believe are 3-5 most valuable tutorials and then detail them as issues and have contributors (hacktoberfest or otherwise) help us in creating them?

**Reply by RafaelGSS** [reply]
Agree!

**Comment by delvedor** [comment]
How about rename for the entire duration of Hacktoberfest the "good first issue" label to "hacktoberfest"?

**Reply by mcollina** [reply]
too much work!

**Reply by delvedor** [reply]
It should be easy, you just need to rename it in https://github.com/fastify/fastify/labels.
It will work only for the main repo, but better than nothing :D

**Reply by jsumners** [reply]
Eh, that wouldn't be clear to people who have no idea what "Hacktoberfest" is.

**Reply by Eomm** [reply]
I have opened a feature request here
https://github.com/Eomm/hacktoberfestify/issues/6

Not much time to work on it right now

**Reply by RafaelGSS** [reply]
Sorry @Eomm. I added the label before saw this. However, I feel that some `good first issue` does not have a low level barrier to new contributors (I know that probably these issues need an update)

**Comment by jsumners** [comment]
I added the label. 🤷‍♂️

**Reply by Ethan-Arrowood** [reply]
🎃

**Reply by RafaelGSS** [reply]
I'm adding good first issue as hacktoberfest as well.

**Comment by Eomm** [comment]
Sorry if I'm offtopic, but maybe the discussion section could be the right place.

`good-first-issue` issues has been always there and always will be.

Why a t-shirt (or a tree for this year) motivate people to contribute more or even start?
Could be the joi to be part of a named event or a community?

What the OSS world (or just the Fastify org) could learn from this event to motivate people to jump in more often?

I think it could be fun to find out that if we put people together, in small teams too, it could lead to a stronger feeling to be part of the community.
I'm asking myself if there is something out there that could improve our discord channel to creates these connections between peers

**Reply by Ethan-Arrowood** [reply]
its crazy what people will do for a free tshirt... jokes aside though, motivation is very tricky and users are unpredictable. I know I definitely enjoyed participating when I was new to open-source; it probably played a part in getting me to the contributor level I'm at today. 

Another event I've enjoyed is the Node.js contributor events they would hold at Node.js Interactive (now OpenJS World). Essentially, maintainers would triage a ton of issues ahead of time and then we gather a bunch of people together to contribute all at the same time; and while they are doing so, the maintainers are walking around helping out in anyway necessary (help with git, help with code, etc). Unfortunately, considering how many of us here are already super busy, this sort of event would be difficult to pull off. 

> What the OSS world (or just the Fastify org) could learn from this event to motivate people to jump in more often?

To answer this more directly, I think the goal of these events is to get developers involved with open source so that hopefully they find projects they enjoy and can continue to contribute too. Additionally, it teaches them how to contribute so when they are using open source tech and run into a bug; they are equipped with the know how to send a PR to fix it rather than just opening an issue.

**Reply by RafaelGSS** [reply]
I feel the event like Hacktoberfest is an open gate for people who think that contributing to open source is an arduous task.

> What the OSS world (or just the Fastify org) could learn from this event to motivate people to jump in more often?

This is a hard question because we don't have feedback from new contributors and because they probably won't feel the same. I think that we can use these events to improve our entry barrier.

**Reply by Eomm** [reply]
I'm asking myself if this is a bad idea:

A maintainer could:

- choose an issue to do
- choose when doing it
- let the fastest people book the session
- start a peer to peer session where the maintainer accomplish the chosen task and the people will listen and ask during the job

The maintainer would do the job as always, but with an interactive public that can be involved

Is this just like streaming? 🤔 
I don't know if I would do this actually 🎃

**Comment by mcollina** [comment]
How can we deal with all the spam PRs that are coming?

**Reply by Eomm** [reply]
With a default reply unfortunately.

From
https://hacktoberfest.digitalocean.com/faq/
https://hacktoberfest.digitalocean.com/details#spam

> We dislike seeing spam pull requests just as much as you, so please give them an invalid or spam label and close them. Pull requests that contain a label with the word invalid or spam won’t be counted toward Hacktoberfest.

**Reply by StarpTech** [reply]
Hopefully, we won't run into https://blog.domenic.me/hacktoberfest/ 😄

**Reply by Ethan-Arrowood** [reply]
There is a method the React team implemented where they limit PR's to the repo to people who have already contributed >3 days ago or something like that

**Reply by jsumners** [reply]
> There is a method the React team implemented where they limit PR's to the repo to people who have already contributed >3 days ago or something like that

That's counter to participating in this thing. The idea is to get new contributors to contribute.

**Reply by Ethan-Arrowood** [reply]
> The idea is to get new contributors to contribute

Totally agree. I haven't seen anything come fastify's way yet so we probably shouldn't do this (or something else restrictive) unless we start getting spammed.

**Reply by jsumners** [reply]
> > The idea is to get new contributors to contribute
> 
> Totally agree. I haven't seen anything come fastify's way yet so we probably shouldn't do this (or something else restrictive) unless we start getting spammed.

We have been getting spammed in unrelated repos. Mostly the website repo. I don't know why because the label doesn't even exist over there.

**Reply by Ethan-Arrowood** [reply]
> Mostly the website repo.

Y'all have been closing these so fast that I haven't even seen it pop up! I wonder if theres a way to restrict PR's to only be openable if they are linked to an issue? This at least may deter the spam and still allow genuine contributions? (but if this is not a feature GitHub supports yet then just ignore me)

**Reply by RafaelGSS** [reply]
My guess would be that because the language on fastify/website is mostly HTML this helps in beginners' filters

**Comment by mcollina** [comment, marked as answer]
Should we opt in to Hacktoberfest? https://hacktoberfest.digitalocean.com/hacktoberfest-update

**Reply by Eomm** [reply]
I would add the topic, to the effort is less than add another label

**Comment by RafaelGSS** [comment]
Re-opening on the next year. We received some new contributors (don't know if are because of the hacktoberfest).

See ya :tada:
